File: cpp/src/gettoken.cpp
#include "gettoken.h"
#include "getnext.h"

void gettoken(void)
{
	nonewcontrolsequence = false;
	getnext();
	nonewcontrolsequence = true;
	if (curcs == 0)
		curtok = curcmd*0x1'00+curchr;
	else
		curtok = cs_token_flag+curcs;
}

File: split/gettoken.pas
procedure gettoken;
begin
  nonewcontrolsequence := false;
  getnext;
  nonewcontrolsequence := true;
  if curcs = 0 then
    curtok := (curcmd * 256) + curchr
  else
    curtok := 4095 + curcs;
end;
